SwiftReflector/MarshalEngine.cs

```diff
@@ -1229,6 +1240,90 @@ CSBaseExpression MarshalProtocolConstrained (BaseDeclaration typeContext, Functi
+			identifiersUsed.Add (proxyPtrID.Name);
+			var proxyIntPtrID = new CSIdentifier (Uniqueify ("proxyIntPtr", identifiersUsed));
+			identifiersUsed.Add (proxyIntPtrID.Name);
+
+			// proxy declaration
+			var isProxyTypeTest = new CSBinaryExpression (CSBinaryOperator.Is, p.Name, new CSIdentifier (csProxyType.ToString ()));
+			var makeProxy = new CSFunctionCall ("SwiftProtocolTypeAttribute.MakeProxy", false, csProxyType.Typeof (), new CSCastExpression (csInterfaceType, p.Name));
+			var proxyExpr = new CSTernary (isProxyTypeTest, new CSCastExpression (csInterfaceType, p.Name), makeProxy, false);
+			var proxyDecl = CSVariableDeclaration.VarLine (proxyID, proxyExpr);
+			preMarshalCode.Add (proxyDecl);
+
+			var proxyCount = new CSFunctionCall ("StructMarshal.RetainCount", false, new CSCastExpression (new CSSimpleType ("ISwiftObject"), proxyID));
+			// proxyRefCount declaration
+			var proxyRefCountDecl = CSVariableDeclaration.VarLine (proxyRefCountID, proxyCount);
+			preMarshalCode.Add (proxyRefCountDecl);
+
+			// proxyPtr decl
+			var proxyPtrDecl = CSVariableDeclaration.VarLine (proxyPtrID, CSArray1D.New (CSSimpleType.Byte, true, new CSIdentifier ("IntPtr.Size")));
+			preMarshalCode.Add (proxyPtrDecl);
+
+			// proxyIntPtr decl
+			var proxyIntPtrDecl = CSVariableDeclaration.VarLine (proxyIntPtrID, new CSFunctionCall ("IntPtr", true, proxyPtrID));
+			preMarshalCode.Add (proxyIntPtrDecl);
+
+			// proxy marshal
+			var proxyMarshal = CSFunctionCall.FunctionCallLine ("StructMarshal.Marshaler.ToSwift", false, proxyID, proxyIntPtrID);
+			preMarshalCode.Add (proxyMarshal);
+
+			// post marshal cleanup
+
+			var condition = new CSBinaryExpression (CSBinaryOperator.And, isProxyTypeTest,
+				new CSBinaryExpression (CSBinaryOperator.GreaterEqual, proxyRefCountID, proxyCount));
+			var disposeCall = CSFunctionCall.FunctionCallLine ($"((ISwiftObject){proxyID.Name}).Dispose", false);
+			var body = CSCodeBlock.Create (disposeCall);
+			var ifPost = new CSLine (new CSIfElse (condition, body), false);
+			postMarshalCode.Add (ifPost);
+
+
+			return proxyIntPtrID;
+
+		}
+
+		CSSimpleType BuildGenericInterfaceFromAssociatedTypes (CSSimpleType interfaceType, ProtocolDeclaration protocol)
+		{
+			
+			return new CSSimpleType (OverrideBuilder.AssociatedTypeProxyClassName (protocol), false, interfaceType.GenericTypes);
+		}
+
+		ProtocolDeclaration ProtocolConstraintFromFunctionDeclaration (NamedTypeSpec swiftType, FunctionDeclaration funcDecl)
+		{
+			var depthIndex = funcDecl.GetGenericDepthAndIndex (swiftType);
+			var genDecl = funcDecl.GetGeneric (depthIndex.Item1, depthIndex.Item2);
+			var constraint = genDecl.Constraints [0] as InheritanceConstraint;
+			var entity = typeMapper.GetEntityForTypeSpec (constraint.InheritsTypeSpec);
+			if (entity == null)
+				throw ErrorHelper.CreateError (ReflectorError.kTypeMapBase + 46, $"Unable to find entity for type {constraint.Inherits}");
```

Comment:
That 46.. why? Should we add it to a constant with a self explanatory name?

Reply:
All errors in BTfS are broken up into larger groups. The number is within that category. Leaving the number in makes it easier to find an error when reported at the UI level (from a customer for example). There is a script called `dora` that will show you the file/line where an error happened which won't work if it's a constant.